### DISCUSSION TITLE
How can I create a two column layout that works both in pptx as well as docx output?

### DISCUSSION THREAD

**Discussion by b-a0** [quarto-dev/quarto-cli#10882] [Q&A]
### Description

I would like to render a Markdown file both to PPTX slides and to a DOCX document.  
However, I cannot find a way to force the use of multiple columns that works in both document formats.

If I run `quarto render .\test.qmd` on below file

```qmd
---
title: "Untitled"
format: 
  docx: default
  pptx: default
---

# Using .columns div
:::: {.columns}
::: {.column}
Left content
:::
::: {.column}
Right content
:::
::::

# Using layout-ncol=2
::: {layout-ncol=2}
Left contents

Right contents
:::
```

I get the Word document on the left and the Powerpoint document on the right:

![image](https://github.com/user-attachments/assets/c7a3100c-1584-4d85-b1a2-0c3b44c3a503)

Word seems to only understand the `{layout-ncol=2}` while Powerpoint only gets the `{{.columns}}` command. The latter also generates this warning when being generated by quarto

```log
WARNING (C:\ProgramData\chocolatey\lib\quarto\tools\share\filters\main.lua:18257) No renderer for PanelLayout
WARNING (C:\ProgramData\chocolatey\lib\quarto\tools\share\filters\main.lua:18262) Don't know how to render PanelLayout without a float; will return empty output
```

How can I create a single document element that is rendered with two columns in both Word and Powerpoint?

I am on Windows 11 23H2 with quarto 1.5.57.

**Comment by mcanouil** [comment]
`columns` is a native Pandoc feature which comes mostly for presentation and has limitations (https://pandoc.org/MANUAL.html#columns).
`layout*` is a Quarto feature which also has some limitations.

Note that Word and PowerPoint documents are the least supported formats in Pandoc (thus also in Quarto) mostly because of the use of two different OpenXML structure.

Here, you can put your content in an external "hidden" (filename prefixed with `_`) Quarto documents that you include via the include shortcode, in combination with conditional contents.
- https://quarto.org/docs/authoring/includes.html
- https://quarto.org/docs/authoring/conditional.html
